Library/Homebrew/development_tools.rb

```diff
@@ -30,6 +30,8 @@ def default_compiler
       # over the system's GCC 4.0
       when /^gcc-4.0/ then gcc_42_build_version ? :gcc : :gcc_4_0
       when /^gcc/ then :gcc
+      # /usr/bin/gcc is symlinked to hardened-cc on Ubuntu
+      when /^hardened-cc/ then :gcc
       else :clang
```

Comment:
An alternative to this might be ```else (OS.mac? ? :clang : :gcc)```.

Part of me prefers that over special-casing in `hardening-wrapper`, but I'm hesitant to make such a big change when this fixes it.